packages/react-components/src/components/VideoGallery.test.tsx

```diff
@@ -156,6 +156,129 @@ describe('VideoGallery default layout', () => {
   });
 });
 
+describe('VideoGallery floating local video layout', () => {
+  beforeAll(() => {
+    registerIcons({
+      icons: {
+        horizontalgalleryleftbutton: <></>,
+        horizontalgalleryrightbutton: <></>
+      }
+    });
+
+    // Need to mock this because the HorizontalGallery uses this function. But JSDOM does not actually do any
+    // rendering so getComputedStyle(document.documentElement).fontSize won't actually have a value
+    jest.spyOn(acs_ui_common, '_convertRemToPx').mockImplementation((rem: number) => {
+      return rem * 16;
+    });
+    // Need to mock hook _useContainerWidth because the returned width is used by HorizontalGallery to decide
+    // how many tiles to show per page
+    jest.spyOn(responsive, '_useContainerWidth').mockImplementation(() => 500);
+  });
+
+  test('should have floating local video tile present', () => {
+    const localParticipant = createLocalParticipant({
+      videoStream: { isAvailable: false, renderElement: createVideoDivElement() }
+    });
+
+    const root = mountVideoGalleryWithLocalParticipant({ localParticipant });
+
+    root.setProps({ layout: 'floatingLocalVideo' });
+    const remoteParticipants = Array.from({ length: 10 }, () =>
+      createRemoteParticipant({
+        videoStream: { isAvailable: false, renderElement: createVideoDivElement() }
+      })
+    );
+    root.setProps({ remoteParticipants });
```

Comment:
In all of these, do you not need to wrap the `root.setProps` with `act`?
We have a bunch of `useEffect` blocks in the implementation, so I'd expect to see a warning in the console if you don't use `act` (and you're not quite testing the right thing without all those `useEffect` hooks getting a chance to execute).

https://reactjs.org/docs/test-utils.html#act